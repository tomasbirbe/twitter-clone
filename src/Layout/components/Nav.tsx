import React from 'react';

// Icons

import {
  HiOutlineHome,
  HiOutlineHashtag,
  HiOutlineBell,
  HiOutlineMail,
  HiOutlineBookmark,
  HiOutlineMenuAlt2,
  HiOutlineUser,
  HiOutlineDotsCircleHorizontal,
  HiDotsHorizontal,
} from 'react-icons/hi';

// Chakra-ui

import { Stack, Text } from '@chakra-ui/layout';
import { Button } from '@chakra-ui/button';
import Icon from '@chakra-ui/icon';
import NavLink from './NavLink';
import TwButton from '../../components/TwButton';
import HomeButton from './HomeButton';
import ProfilePic from '../../components/ProfilePic';

const Nav = () => (
  <Stack
    as="nav"
    borderInlineEnd="1px"
    borderColor="gray.100"
    height="100%"
    width="260px"
    justifyContent="space-between"
    paddingBlock={2}
  >
    <Stack
      as="ul"
      paddingInlineEnd={8}
      justify="space-between"
      alignItems="flex-start"
      spacing={3}
    >
      <HomeButton />
      <NavLink icon={HiOutlineHome} text="Home" />
      <NavLink icon={HiOutlineHashtag} text="Explore" />
      <NavLink icon={HiOutlineBell} text="Notifications" />
      <NavLink icon={HiOutlineMail} text="Messages" />
      <NavLink icon={HiOutlineBookmark} text="Bookmarks" />
      <NavLink icon={HiOutlineMenuAlt2} text="Lists" />
      <NavLink icon={HiOutlineUser} text="Profile" />
      <NavLink icon={HiOutlineDotsCircleHorizontal} text="More" />
      <Stack width="100%" paddingBlockStart={2}>
        <TwButton
          fontSize={18}
          color="white"
          bg="blue.500"
          paddingBlock={7}
          _hover={{
            bg: 'blue.600',
          }}
        >
          Tweet

        </TwButton>
      </Stack>

    </Stack>

    <Stack
      paddingInlineEnd={3}
    >
      <Button
        borderRadius="full"
        bg="transparent"
        paddingInline={2}
        paddingBlock={8}
        _hover={{
          bg: 'gray.200',
        }}
      >
        <Stack
          direction="row"
          align="center"
          justify="flex-start"
          width="100%"
        >
          <Stack>
            <ProfilePic size={10} />
          </Stack>
          <Stack
            justify="space-around"
            align="flex-start"
            spacing={1}
          >
            <Text>Kitten</Text>
            <Text fontSize={14} color="gray.500">@CatLover</Text>
          </Stack>
        </Stack>
        <Stack>
          <Icon as={HiDotsHorizontal} boxSize={4} />
        </Stack>
      </Button>
    </Stack>
  </Stack>

);

export default Nav;
