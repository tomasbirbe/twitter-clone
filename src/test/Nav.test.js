import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import Nav from '../Layout/components/Nav';

configure({ adapter: new Adapter() });

describe('Testing the first app', () => {
  it('should test the first component', () => {
    const component = shallow(<Nav />);
    expect(component).toMatchSnapshot();
  });
});
