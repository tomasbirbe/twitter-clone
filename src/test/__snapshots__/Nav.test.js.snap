// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the first app should test the first component 1`] = `ShallowWrapper {}`;
